### PR TITLE
Add upgrade test for LIKE and Use optimisation for Const node for AI collations

### PR DESCRIPTION
**Pull request by ahmed-shameem**
### Description

This commit introduces the upgrade tests for LIKE operator for CI_AI and CS_AI collations. Tests for remove_accents_internal has also been added by creating function, view, stored procedure which invokes remove_accents_internal. 

This change also updates the capacity to 4 * len(input_string) as 4 is the max number of bytes which can be allocated for a single character in ICU ([https://www.rfc-editor.org/rfc/rfc3629#section-3](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)). 

As we are modifying the left and right nodes of the LIKE operator to FuncExpr, we lose the optimisation for CI_AI which was done for CI_AS for constant prefix in pattern. This commit handles the Const node separately so that we can perform the optimization. 

### Issues Resolved

TASK : BABEL-4791
Signed-off-by: Shameem Ahmed [shmeeh@amazon.com](mailto:shmeeh@amazon.com)

### Test Scenarios Covered ###
* **Use case based -**

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/collation.c [hunk 1]
```diff
@@ -485,7 +508,12 @@ Datum remove_accents_internal(PG_FUNCTION_ARGS)
 	pfree(input_str);
 
 	limit = len_uinput;
-	capacity = MaxAttrSize / sizeof(char);
+	/* 
+	 * set the capacity to limit * MAX_BYTES_PER_CHAR if it is less than INT32_MAX
+	 * else set it to INT32_MAX as capacity is of int32_t datatype so it can 
+	 * has maximum INT32_MAX value
```
Comment (shah-nirmit) [inline contrib/babelfishpg_tsql/src/collation.c]: Nit: have

File: contrib/babelfishpg_tsql/src/collation.c [hunk 2]
```diff
@@ -485,7 +508,12 @@ Datum remove_accents_internal(PG_FUNCTION_ARGS)
...
+	 */
+	capacity = limit * MAX_BYTES_PER_CHAR < INT32_MAX ? limit * MAX_BYTES_PER_CHAR : INT32_MAX;
```
Comment (shah-nirmit) [inline contrib/babelfishpg_tsql/src/collation.c]: Nit: enclose within ()

**Review by shah-nirmit — APPROVED** [review]

File: contrib/babelfishpg_tsql/src/collation.c (hunk 1 above)
Comment (ahmed-shameem) [reply contrib/babelfishpg_tsql/src/collation.c]: Updated

File: contrib/babelfishpg_tsql/src/collation.c (hunk 2 above)
Comment (ahmed-shameem) [reply contrib/babelfishpg_tsql/src/collation.c]: Updated

File: contrib/babelfishpg_tsql/src/collation.c [hunk 3]
```diff
@@ -485,7 +508,12 @@ Datum remove_accents_internal(PG_FUNCTION_ARGS)
 	pfree(input_str);
 
 	limit = len_uinput;
-	capacity = MaxAttrSize / sizeof(char);
+	/* 
+	 * set the capacity to limit * MAX_BYTES_PER_CHAR if it is less than INT32_MAX
+	 * else set it to INT32_MAX as capacity is of int32_t datatype so it can 
+	 * have maximum INT32_MAX value
+	 */
+	capacity = ((limit * MAX_BYTES_PER_CHAR) < INT32_MAX) ? (limit * MAX_BYTES_PER_CHAR) : INT32_MAX;
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/src/collation.c]: 1. There is already macro defined by Postgres, we should use it`PG_UINT32_MAX`. 
2. capcity and other size related params should be unsigned int. 
3. How will this overflow check work?

**Review by Deepesh125 — APPROVED** [review]

**Pull request merged**